### PR TITLE
FrontendLogging: A very basic frontend logging lib

### PR DESCRIPTION
**Pull request by torkelo**
This explores a super basic logging frontend lib that would allow us to add some debug logging to complex & async processes and enable in both dev & prod builds. 

Goals
* [x] As lightweight as possible with minimal runtime perf impact when disabled
* [x] Support hierarchical loggers & level config 
* [x] easy runtime config change (Example:  `window.grafanaLogging.setLoggerLevel("navigation", "debug"); ` console 
* [x] Support for multiple appenders (So we can in the future add a custom appender to include these in Inspect/Help tabs or support bundles)  
* [x] Skip levels per appender for now 

Future additions
* [ ]  Enable loggers via URL flags (important if we want to capture a process that only happens on first load)

### CODE REVIEW

File: packages/grafana-data/src/logging/Appenders.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+export class ConsoleAppender implements LogAppender {
+  append(logger: Logger, level: LogLevel, args: unknown[]): void {
+    if (level === LogLevel.Debug) {
+      // eslint-disable-next-line no-console
+      console.debug(`[${logger.name}]`, ...args);
+    }
+  }
+}
```
Comment (torkelo) [inline packages/grafana-data/src/logging/Appenders.ts:29]: super basic implementation missing log levels, want to get some feedback before fleshing it out

File: public/app/features/dashboard/dashgrid/PanelStateWrapper.tsx [hunk 2]
```diff
@@ -624,6 +628,10 @@ export class PanelStateWrapper extends PureComponent<Props, State> {
     const { errorMessage, data } = this.state;
     const { transparent } = panel;
 
+    if (panelLogger.isDebugEnabled()) {
```
Comment (jarben) [inline public/app/features/dashboard/dashgrid/PanelStateWrapper.tsx:631]: Sorry why this IF? Isn't this handled by appenders?

**Comment by jarben** [comment]
This looks really nice! 

One thing I'd consider adding in the future is log formatters so that we can easily turn on/off timestamps etc. Winstonjs has a separate and reasonably lightweight package for this: https://github.com/winstonjs/logform which does stuff like: 
- timestamps 
- ms between logs
- colours 
- JS errors
- ...

File: public/app/features/dashboard/dashgrid/PanelStateWrapper.tsx (hunk 2 above)
Comment (scottlepp) [reply public/app/features/dashboard/dashgrid/PanelStateWrapper.tsx:631]: might be nice for the first arg to be a string or function.  then you don't need an if statement for doing costly string building

Comment (torkelo) [reply public/app/features/dashboard/dashgrid/PanelStateWrapper.tsx:631]: @scottlepp yea, was thinking of that as well. perf cost of creating a function should be low enough for that to be preferable to this that requires 3 lines of code

**Pull request closed without merging**